File: O.cpp
#include <iostream>
#include <vector>
#include <map>
#include <algorithm>

using namespace std;


int dist(vector<int> a){
    sort(a.begin(), a.end());
    int cur_sum = 0, ans = 0;
    for (int i = 0; i < a.size(); i++){
        cur_sum += a[i];
        ans += a[i] * (i + 1) - cur_sum;
    }
    return 2 * ans;
}


int main() {
    int k, n;
    cin >> k;
    cin >> n;
    vector<int> xs, ys;
    map< int, vector<int> > clusters;
    for (int i = 0; i < n; i++){
        int x, y;
        cin >> x >> y;
        xs.push_back(x);
        ys.push_back(y);
    }
    int all_distance = dist(xs);
    for (int i = 0; i < n; i++){
        clusters[ys[i]].push_back(xs[i]);
    }
    int inner_distance = 0;
    for (auto i: clusters){
        inner_distance += dist(i.second);
    }
    cout << inner_distance << endl;
    cout << all_distance - inner_distance << endl;
    return 0;

}

File: O.go
package main

import "fmt"
// import "math"
import "bufio"
import "os"
import "sort"


func Dist(array []int) int64 {
    sort.Ints(array)
    var summ int64
    var ans int64
    summ = 0
    ans = 0
    for i := 1; i < len(array) + 1; i++{
        summ += int64(array[i - 1])
        ans += int64(array[i - 1]) * int64(i) - summ
    }
    return 2 * ans
}


func main(){
    var n, k int
    in := bufio.NewReader(os.Stdin)
    fmt.Fscan(in, &k)
    fmt.Fscan(in, &n)
    xs := make ([]int, n)
    ys := make ([]int, n)
    for i := 0; i < n; i++{
        fmt.Fscan(in, &xs[i], &ys[i])
    }
    clusters := make (map[int] []int)

    for i := 0; i < n; i++{
        clusters[ys[i]] = append(clusters[ys[i]], xs[i])
    }
//     fmt.Println(clusters)

    var inner_summ int64
    inner_summ = 0
    for _, value := range clusters {
        inner_summ += Dist(value)
    }
    fmt.Println(inner_summ)
    all_dist := Dist(xs)
    outer_summ := all_dist - inner_summ
    fmt.Println(outer_summ)

}